pkg/kubernetes/reconcile.go

```diff
@@ -52,21 +52,29 @@ var kindOrder = []string{
 	"APIService",
 }
 
+const (
+	MetadataPrefix   = "tanka.dev"
+	LabelEnvironment = MetadataPrefix + "/environment"
+)
+
 // Reconcile extracts kubernetes Manifests from raw evaluated jsonnet <kind>/<name>,
 // provided the manifests match the given regular expressions. It finds each manifest by
 // recursively walking the jsonnet structure.
 //
 // In addition, we sort the manifests to ensure the order is consistent in each
 // show/diff/apply cycle. This isn't necessary, but it does help users by producing
 // consistent diffs.
-func Reconcile(raw map[string]interface{}, spec v1alpha1.Spec, targets []*regexp.Regexp) (state manifest.List, err error) {
+func Reconcile(raw map[string]interface{}, cfg v1alpha1.Config, targets []*regexp.Regexp) (state manifest.List, err error) {
 	extracted, err := extract(raw)
 	if err != nil {
 		return nil, errors.Wrap(err, "flattening manifests")
 	}
 
 	out := make(manifest.List, 0, len(extracted))
 	for _, m := range extracted {
+		// inject tanka.dev/environment label
+		m.Metadata().Labels()[LabelEnvironment] = cfg.Metadata.NameLabel()
```

Comment:
This needs to be behind a feature flag or enabled through the tanka Config.
We dont want labels to suddenly start being added as soon as one person updates their tanka version.

Reply:
done